### PR TITLE
[pcapplusplus] update to 23.09

### PR DESCRIPTION
**Pull request by clementperon**
Update PcapPlusPlus to 23.09. This new version use CMake Build System and is much more easier to integrate with vcpkg.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by clementperon** [comment]
@microsoft-github-policy-service agree

File: versions/baseline.json [hunk 1]
```diff
@@ -6405,7 +6405,7 @@
       "port-version": 9
     },
     "pcapplusplus": {
-      "baseline": "22.11",
+      "baseline": "23.9",
```
[the remark this answers is not in the document]
Comment (clementperon) [reply versions/baseline.json:6408]: It should be server compatible. So not possible AFAIK

[the remark this answers is not in the document]
Comment (seladb) [reply versions/baseline.json:6408]: It used to have leading zeros in previous versions 🤔 : https://github.com/microsoft/vcpkg/commit/d1135a5fe2098c4048f96e72808b4bff77478a4d

[the remark this answers is not in the document]
Comment (clementperon) [reply versions/baseline.json:6408]: It was a version string not a version. I will check thanks

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

File: ports/pcapplusplus/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pcapplusplus",
-  "version": "22.11",
+  "version": "23.09",
```
Comment (dg0yt) [inline ports/pcapplusplus/vcpkg.json:3]: No, there can't be leading zeros in version scheme `version`, even if vcpkg reviewer suggest it.
```suggestion
  "version": "23.9",
```

Comment (clementperon) [reply ports/pcapplusplus/vcpkg.json:3]: Can I change it to version-string ?

Comment (dg0yt) [reply ports/pcapplusplus/vcpkg.json:3]: `version-string` doesn't allow for `version>=`.

File: ports/pcapplusplus/portfile.cmake [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
-endif()
+# At the moment PcapPlusPlus doesn't support dynamic linkage on Windows
+set(VCPKG_LIBRARY_LINKAGE static)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO seladb/PcapPlusPlus
-    REF v22.11
-    SHA512 41a507ce385d8549186eeec1a1ae138070ab2021d9ffd907829551b3b865ecb526fa05a0ff9ca01b41a2a2807a60a3cba016f62063d30d849282c83e17a2b6e1
+    REF "v${VERSION}"
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/pcapplusplus/portfile.cmake:7]: Port  `usd` does this:
~~~cmake
string(REGEX REPLACE "^([0-9]+)[.]([0-9])\$" "\\1.0\\2" USD_VERSION "${VERSION}")
...
    REF "v${USD_VERSION}"
~~~

[the remark this answers is not in the document]
Comment (clementperon) [reply ports/pcapplusplus/portfile.cmake]: Thanks very helpfull :)

File: ports/pcapplusplus/vcpkg.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pcapplusplus",
-  "version": "22.11",
+  "version": "23.9",
```
[the remark this answers is not in the document]
Comment (clementperon) [reply ports/pcapplusplus/vcpkg.json:3]: Not valid semver syntax, I kept 23.09 and leading zero with Regex syntax

File: ports/pcapplusplus/portfile.cmake [hunk 5]
```diff
@@ -1,29 +1,30 @@
-if (VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
-endif()
+# At the moment PcapPlusPlus doesn't support dynamic linkage on Windows
+set(VCPKG_LIBRARY_LINKAGE static)
```
Comment (dg0yt) [inline ports/pcapplusplus/portfile.cmake:2]: Why did you change these lines if it is still a windows-only limitation?

File: ports/pcapplusplus/portfile.cmake [hunk 6]
```diff
@@ -1,29 +1,30 @@
...
-file(COPY "${CURRENT_PORT_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DPCAPPP_BUILD_EXAMPLES=OFF
 )
 
 vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME PcapPlusPlus CONFIG_PATH lib/cmake/pcapplusplus)
```
Comment (dg0yt) [inline ports/pcapplusplus/portfile.cmake:22]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/pcapplusplus)
```

PACKAGE_NAME is a misnomer: It is used for the target *directory* name. We prefer lower-case dir names,  and the default is `${PORT}`.

**Comment by clementperon** [comment]
@LilyWangLL I have pushed a commit to fix @dg0yt comments.

**Review by LilyWangLL — APPROVED** [review]
